test/e2e/dataplane/tcp_pod_to_service.go

```diff
@@ -30,29 +30,34 @@ var _ = Describe("[dataplane] Basic Pod to Service tests across clusters without
 	})
 })
 
-func testPod2ServiceTCP(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling) {
+func TestPod2ServiceTCP(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling, listenerPodCluster framework.ClusterIndex, connectorPodCluster framework.ClusterIndex) {
 
-	listenerPod, connectorPod := runAndVerifyNetworkPod2ServicePair(f, leftScheduling, rightScheduling)
+	listenerPod, connectorPod := runAndVerifyNetworkPod2ServicePair(f, leftScheduling, rightScheduling, listenerPodCluster, connectorPodCluster)
 
 	By("Verifying what the pods sent to each other contain the right UUIDs")
 	Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Config.Data))
 	Expect(connectorPod.TerminationMessage).To(ContainSubstring(listenerPod.Config.Data))
 }
 
 func testPod2ServiceTCPIPPreservation(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling) {
-
-	listenerPod, connectorPod := runAndVerifyNetworkPod2ServicePair(f, rightScheduling, leftScheduling)
-
+	listenerPod, connectorPod := runAndVerifyNetworkPod2ServicePair(f, rightScheduling, leftScheduling, framework.ClusterB, framework.ClusterA)
 	framework.Logf("Connector pod has IP: %s", connectorPod.Pod.Status.PodIP)
 	By("Verifying the output of listener pod which must contain the source IP")
 	Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Pod.Status.PodIP))
 }
 
-func runAndVerifyNetworkPod2ServicePair(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling) (*framework.NetworkPod, *framework.NetworkPod) {
+func TestPod2ServiceTCPNoConnection(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling, listenerPodCluster framework.ClusterIndex, connectorPodCluster framework.ClusterIndex) {
+	runAndVerifyNetworkPod2ServicePair(f, leftScheduling, rightScheduling, listenerPodCluster, connectorPodCluster)
+	By("Verifying that the connection attempt from connector pod errors out")
+	//TODO Determine what to "Expect"
+	Expect("Error to have occured")
+}
```

Comment:
This is incomplete and also should be in a separate PR.

Reply:
Removed from here. Will push a separate PR.